File: node.js/src/original/domain/books/entities/reviewer.ts
import { DeepReadonly, NonEmptyString, NonEmptyUuid } from '#core/typing';

export type ReviewerId = NonEmptyUuid<'ReviewerId'>;
export type ReviewerName = NonEmptyString<'ReviewerName'>;

export type Reviewer = DeepReadonly<{ Id: ReviewerId; Name: ReviewerName }>;

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/entities/Reviewer.java
package io.eventdriven.slimdownaggregates.original.domain.books.entities;

import static io.eventdriven.slimdownaggregates.original.infrastructure.validation.Validation.assertNotNull;

public record Reviewer(ReviewerId id, ReviewerName name) {
  public Reviewer {
    assertNotNull(id);
    assertNotNull(name);
  }
}
